bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/utils/MyBMWConfigurationChecker.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mybmw.internal.utils;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.mybmw.internal.MyBMWBridgeConfiguration;
+
+/**
+ * 
+ * checks if the configuration is valid
+ * 
+ * @author Bernd Weymann - Initial contribution
+ * @author Martin Grassl - extracted to own class
+ */
+@NonNullByDefault
+public final class MyBMWConfigurationChecker {
+    public static boolean checkConfiguration(MyBMWBridgeConfiguration config) {
+        if (Constants.EMPTY.equals(config.userName) || Constants.EMPTY.equals(config.password)) {
```

Comment:
isBlank or isEmpty is dangerous as it throws an NPE if the string to be checked is null. In this case as the initial value is "" I changed it

Reply:
What agbout `config.userName == null || config.userName.isBlank()`

Reply:
I'm working a lot with Sonarqube and there is the rule that the string literal should be on the left side of the comparison to avoid null errors, see https://rules.sonarsource.com/java/RSPEC-1132/. So I would not do it this way.

Reply:
and an or comparison would execute both branches, right?

Reply:
> I'm working a lot with Sonarqube and there is the rule that the string literal should be on the left side of the comparison to avoid null errors, see https://rules.sonarsource.com/java/RSPEC-1132/. So I would not do it this way.

Yes, absolutely true. If you don;t this will be picked up code analysis.

> and an or comparison would execute both branches, right?

No, java evaluates them one by one, as soon as one qualify's it will fall throug and it won;t evaluate the others. But here you don;t need to check for null as config has non nullable properties.

Reply:
you're right...